habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/migration/CustomPoetrycoreVersionMigration.java

```diff
@@ -0,0 +1,162 @@
+                    isMinorVersionUpdateRequired = false;
+                    // check if we need to upgrade the poetry-core version.
+                   if(IsPoetrycoreUpgradeRequired(dependency)) {
+                       return true;
+                   }
+                }
+            }
+        }
+        return false;
+    }
+
+    @Override
+    protected boolean performMigration(File pyProjectTomlFile) {
+        String fileContent = StringUtils.EMPTY;
+        try (BufferedReader reader = new BufferedReader(new FileReader(pyProjectTomlFile))) {
+            String line = reader.readLine();
+            while (line != null) {
+                boolean isEmptyLine = line.isBlank();
+
+                if (line.contains(StringUtils.SPACE) && line.contains(TomlUtils.EQUALS)) {
+                    String key = line.substring(0, line.indexOf(StringUtils.SPACE));
+                    if (key == null) {
+                        key = line.substring(0, line.indexOf(TomlUtils.EQUALS));
+                    }
+
+                    if (key != null) {
+                        key = key.strip();
+
+                        TomlReplacementTuple matchedTuple = replacements.get(key);
+                        if ((matchedTuple != null) && (line.contains(TomlUtils.POETRY_CORE)) && (line.contains(TomlUtils.EQUALS))) {
+                            // update the poetry-core major version upgrade if required.
+                            StringBuilder stringBuilder = new StringBuilder(line);
+                            if(isMajorVersionUpdateRequired){
+                                int majorIndex = line.lastIndexOf(TomlUtils.EQUALS) + 1;
+                                int endMajorIndex = line.indexOf(TomlUtils.DOT);
+                                String poetryCoreReqMajorVer = PoetryUtil.getMajorReqVersion(PoetryUtil.POETRY_CORE_VERSION_REQUIREMENT);
+                                stringBuilder = stringBuilder.replace(majorIndex, endMajorIndex, poetryCoreReqMajorVer);
+                                line = stringBuilder.toString();
+                                logger.info("Updating the poetry-core major version.");
+                            }
+
+                            // update the poetry-core minor version if required.
+                            if (isMinorVersionUpdateRequired) {
+                                int minorVersionIndex = line.indexOf(TomlUtils.DOT) + 1;
+                                int endMinorVersionIndex = line.lastIndexOf(TomlUtils.DOT);
+                                String strPoetryCoreReqMinorVersion = PoetryUtil.getMinorReqVersion(PoetryUtil.POETRY_CORE_VERSION_REQUIREMENT);
+                                stringBuilder = stringBuilder.replace(minorVersionIndex, endMinorVersionIndex, strPoetryCoreReqMinorVersion);
+                                line = stringBuilder.toString();
+                                logger.info("Updating the Poetry-core major version to {}.", strPoetryCoreReqMinorVersion);
```

Comment:
A: Typo? The string refers to the major version even though this is for the minor. 

Also, would the logs on line 99 and 89 be better suited for logger.debug? Doesn't make sense to say it's upgrading twice in the info logs every time; I think the single info statement on line 101 is the only one helpful during a normal build.